content/doc/developer/security/csp.adoc

```diff
@@ -0,0 +1,110 @@
+---
+title: Content-Security-Policy Compatibility
+layout: developerguide
+---
+
+== Introduction
+
+From https://content-security-policy.com/[content-security-policy.com]:
+
+> _Content-Security-Policy_ is the name of a HTTP response header that modern browsers use to enhance the security of the document (or web page). The Content-Security-Policy header allows you to restrict how resources such as JavaScript, CSS, or pretty much anything that the browser loads.
+
+Using Content-Security-Policy (CSP), injection attacks like cross-site scripting can be prevented.
+
+// TODO Keep up to date with versions
+Unfortunately, as of Jenkins 2.360, the Jenkins classic UI is not yet compatible with the CSP directives that would allow preventing such injection attacks.
+
+This guide documents how to write UI code in a manner that is expected to be compatible with future use of CSP directives on the Jenkins UI.
+
+
+
+== Guidelines
+
+
+=== Inline JavaScript blocks
+
+Do not use inline JavaScript (JS) on the Jenkins UI, i.e., JS embedded in HTML output.
+
+This is typically done with `<script>` tags, like so:
+
+[source, html]
+<script type="text/javascript">
+alert("Hello, world!");
+</script>
+
+The guidelines in link:/doc/developer/security/xss-prevention/#passing-values-to-javascript[the documentation on XSS prevention] can be useful to pass arguments to JavaScript, or otherwise control its behavior dynamically.
+
+You can generally use https://github.com/jenkinsci/stapler/blob/master/docs/jelly-taglib-ref.adoc#adjunct[Stapler adjuncts] to load files related to UI views and ensure they are loaded only once.
+
+Exmples of this are: https://github.com/jenkinsci/jenkins/pull/6849[jenkinsci/jenkins#6849]
+
+
+=== Inline event handlers
+
+Event handlers like `onclick` or `onblur` would be defined in separate files.
+
+For this to work, the element that would have the inline event handler attribute(s) would need a class or ID by which it can be looked up from JS.
```

Comment:
```suggestion
For this to work, the element that would have had the inline event handler attribute(s) needs a class or ID by which it can be looked up from JS.
```